src/navigation/reroute.js

```diff
@@ -79,10 +79,20 @@ export function reroute(pendingPromises = [], eventArguments) {
 
   function performAppChanges() {
     return Promise.resolve().then(() => {
+      // https://github.com/single-spa/single-spa/issues/545
+      window.dispatchEvent(
+        new CustomEvent(
+          appsThatChanged.length === 0
+            ? "single-spa:before-no-app-change"
+            : "single-spa:before-app-change",
+          getCustomEventDetail(true)
+        )
+      );
+
       window.dispatchEvent(
         new CustomEvent(
           "single-spa:before-routing-event",
-          getCustomEventDetail()
+          getCustomEventDetail(true)
```

Comment:
Technically a breaking change that now this event has a different `detail` property. I doubt people are relying on it, though? The "new" app status was wrong before this change.